parsl/serialize/facade.py

```diff
@@ -0,0 +1,174 @@
+from parsl.serialize.concretes import *  # noqa: F403,F401
+from parsl.serialize.base import METHODS_MAP_DATA, METHODS_MAP_CODE
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class ParslSerializer(object):
+    """ ParslSerializer gives you one interface to multiple serialization libraries
+    """
+    __singleton_instance = None
+
+    def __new__(cls, *args, **kwargs):
+        """ Make ParslSerializer a singleton using the fact that there's only one instance
+        of a class variable
+        """
+        if ParslSerializer.__singleton_instance is None:
+            ParslSerializer.__singleton_instance = object.__new__(cls, *args, **kwargs)
+        return ParslSerializer.__singleton_instance
+
+    def __init__(self):
+        """ Instantiate the appropriate classes
+        """
+        headers = list(METHODS_MAP_CODE.keys()) + list(METHODS_MAP_DATA.keys())
+        self.header_size = len(headers[0])
```

Comment:
this line makes me incredibly uncomfortable. it's assuming that all headers keys are the same length, or something like that. If there's a fixed value that all headers lengths need to be, hard code it.

or if you want to be more fancy, what happens when headers are different lengths?